webapp/channels/src/components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx

```diff
@@ -21,34 +21,32 @@ export interface Props extends PropsFromRedux {
     currentTeamName: string;
 }
 
-export default class SidebarBaseChannel extends React.PureComponent<Props> {
-    handleLeavePublicChannel = (callback: () => void) => {
-        this.props.actions.leaveChannel(this.props.channel.id);
+const SidebarBaseChannel = ({channel, currentTeamName, actions}: Props) => {
+    const intl = useIntl();
+
+    const handleLeavePublicChannel = (callback: () => void) => {
+        actions.leaveChannel(channel.id);
         trackEvent('ui', 'ui_public_channel_x_button_clicked');
         callback();
     };
 
-    handleLeavePrivateChannel = (callback: () => void) => {
-        this.props.actions.openModal({modalId: ModalIdentifiers.LEAVE_PRIVATE_CHANNEL_MODAL, dialogType: LeaveChannelModal, dialogProps: {channel: this.props.channel}});
+    const handleLeavePrivateChannel = (callback: () => void) => {
+        actions.openModal({modalId: ModalIdentifiers.LEAVE_PRIVATE_CHANNEL_MODAL, dialogType: LeaveChannelModal, dialogProps: {channel}});
         trackEvent('ui', 'ui_private_channel_x_button_clicked');
         callback();
     };
 
-    getChannelLeaveHandler = () => {
-        const {channel} = this.props;
-
+    const getChannelLeaveHandler = () => {
```

Comment:
`getChannelLeaveHandler` is ok not being a callback because we are just calling it, but I am not sure if it is even needed. We could have something like:

```
let channelLeaveHandler = null;
if (channel.type === Constants.OPEN_CHANNEL && channel.name !== Constants.DEFAULT_CHANNEL) {
    channelLeaveHandler = handleLeavePublicChannel;
} else if (channel.type === Constants.PRIVATE_CHANNEL) {
    channelLeaveHandler = handleLeavePrivateChannel;
}
```
I am not sure about this, so... what are your thoughts?

Reply:
I agree, the logic is clean and straightforward.